qualtran/bloqs/multiplexers/apply_lth_bloq.py

```diff
@@ -0,0 +1,121 @@
+from qualtran.bloqs.multiplexers.unary_iteration_bloq import UnaryIterationGate
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+from qualtran.symbolics import ceil, log2, prod
+
+
+@frozen
+class ApplyLthBloq(UnaryIterationGate, SpecializedSingleQubitControlledGate):  # type: ignore[misc]
+    r"""A SELECT operation that executes one of a list of bloqs $U_l$ based on a quantum index:
+
+    $$
+    \mathrm{SELECT} = \sum_{l}|l \rangle \langle l| \otimes U_l
+    $$
+
+    This bloq uses the unary iteration scheme to apply `ops[selection]` controlled on an optional single-bit `control` register.
+
+    Args:
+        ops: List of bloqs. Each bloq must have identical registers that are all THRU.
+        selection_regs: List of selection registers, 1D selection index inferred if not provided.
+        control_val: If provided, a singly controlled gate is constructed.
+
+    Registers:
+        selection: The indices of the bloq in `ops` to execute.
+        control: The control bit if specified above.
+        [user_spec]: The output registers of the bloqs in `ops`.
+
+    References:
+        [Encoding Electronic Spectra in Quantum Circuits with Linear T Complexity](https://arxiv.org/abs/1805.03662).
+        Babbush et. al. (2018). Section III.A. and Figure 7.
+    """
+
+    ops: NDArray[Bloq] = field(  # type: ignore[type-var]
+        converter=lambda x: np.array(x) if isinstance(x, Iterable) else x,
+        eq=lambda d: tuple(d.flat),
+    )
+    selection_regs: Optional[Tuple[Register, ...]] = None
+    control_val: Optional[int] = None
+
+    def __attrs_post_init__(self):
+        if cast(int, prod(self.ops.shape)) <= 1:
+            raise ValueError("Must have at least two operations.")
+        if not all(tuple(u.signature) == self.target_registers for u in self.ops.flat):
+            raise ValueError("All ops must have the same signature.")
+        if not all(r.side == Side.THRU for u in self.ops.flat for r in u.signature):
+            raise ValueError("All ops must have only THRU registers.")
+
+    @cached_property
+    def control_registers(self) -> Tuple[Register, ...]:
+        return () if self.control_val is None else (Register('control', QBit()),)
+
+    @cached_property
+    def selection_registers(self) -> Tuple[Register, ...]:
+        if self.selection_regs is None:
+            return tuple(
+                Register(
+                    "selection" if len(self.ops.shape) == 1 else f"selection{i + 1}",
+                    BoundedQUInt(ceil(log2(k)), k),
+                )
+                for i, k in enumerate(self.ops.shape)
+            )
+        return self.selection_regs
+
+    @cached_property
+    def target_registers(self) -> Tuple[Register, ...]:
+        return tuple(self.ops.flat[0].signature)
+
+    def nth_operation_callgraph(self, ssa: SympySymbolAllocator, **kwargs: int) -> Set[BloqCountT]:
+        return self.ops[tuple(kwargs.values())].controlled().build_call_graph(ssa)
```

Comment:
We can just return `{self.ops[tuple(kwargs.values())].controlled()}`  instead of explicitly calling the `build_call_graph` of the subbloq. The call graph of `ApplyLthBloq` should simply decompose into `self.ops` and their subgraph would be inferred automatically by the call graph infrastructure.